design-proposals/persist-vmi-firmware-uuid.md

```diff
@@ -0,0 +1,129 @@
+# Overview
+This proposal introduces a mechanism to persist the firmware UUID of a Virtual Machine Instance (VMI) in KubeVirt.
+By storing the firmware UUID, we ensure that it remains consistent across VMI restarts.
+which is crucial for applications and services that rely on the UUID for identification or licensing purposes.
+
+## Motivation
+By definition, UUID must be a universally-unique identifier.
+The current implementation for the automatically-computed Firmware UUID is not universally unique: 
+Two VMs with the same name get the same UUID, even if they reside in different namespace or different clusters.
+This proposal discusses how to fix this, while ensuring that already-identified VMs keep their current ID.
+
+## Goals
+Generate and persist universally-unique Firmware.UUID for new VMs
+Keep the Firmware.UUID of pre-existing VMs
+
+## Non Goals
+
+
+## Definition of Users
+End Users: Individuals or organizations running VMs and VMIs on KubeVirt who require consistent firmware UUIDs for their applications.
+
+## User Stories
+As an end-user, I expect my VMI to maintain its identity across restarts.
+
+## Repos
+Kubevirt/kubevirt
+
+# Design
```

Comment:
The introduction of a persistent, universally unique firmware UUID can be effectively divided into two key phases:

1. The initial focus on minimizing workload disruption by preserving the current generation method and persisting the UUID, followed by:
2. Transitioning to the generation of a universally unique firmware UUID.

Each phase offers multiple implementation approaches (such as using `namespace+name` for UUID generation or opting for a pseudo-random hash). To ensure clarity and thorough consideration, each phase should be addressed in separate subsections within the Design section.